backend/infrahub/graphql/mutations/main.py

```diff
@@ -134,6 +135,20 @@ async def _refresh_for_profile_update(
             )
         return obj
 
+    @classmethod
+    async def _call_mutate_create_object(cls, data: InputObjectType, db: InfrahubDatabase, branch: Branch):
+        """
+        Wrapper around mutate_create_object to potentially activate locking.
+        """
+        lock_names = registry.schema.get_schema_branch(name=branch.name).get_kind_lock_names_on_object_mutation(
```

Comment:
we should get the schema branch from the `InfrahubDatabase` object : `db.get_schema_branch()`